app/views/admin/loans/print.html.slim

```diff
@@ -9,8 +9,25 @@ section data-container="loan"
       h2 = t('loan.details')
       = render "admin/loans/details", loan: @loan
 
-    section.featured-image
-      = image_tag(@first_image.item.url) if @first_image
+    - if @images.first
+      section.featured-image
+        = image_tag(@images.first.item)
+
+    - unless @image_list.empty?
+      - @image_list.each_with_index do |image, index|
+        / HTML instead of Slim is used in this block to have control over how items are nested
+        / A new page and heading appears every 8 images
+        - if index % 8 == 0
+          <section class="images">
```

Comment:
I see what you're doing here. I appreciate the comment and bravo for figuring this out!

It's best to try to avoid conditionally rendering closing tags if at all possible because it's hard to read and error prone. We are doing something similar to this in `_questionnare_group.html.slim` and it's been a real pain. It's better to set up your model/data structure so that you can iterate over it in a normal way.

If you do as I suggest with `each_slice` above, you'll be able to just iterate over the slices first and then the individual images and you won't have to use HTML.

Please let me know if you try this and it doesn't work or if you need help.

Thanks! And again, nice job working through this and getting a working solution! It's not an easy problem!